vendor/bat-native-ledger/src/bat/ledger/internal/database/database_publisher_prefix_list.cc

```diff
@@ -0,0 +1,204 @@
+
+namespace {
+
+const char kTableName[] = "publisher_prefix_list";
+
+constexpr size_t kHashPrefixSize = 4;
+constexpr size_t kMaxInsertRecords = 100'000;
+
+std::tuple<PrefixIterator, std::string, size_t> GetPrefixInsertList(
+    PrefixIterator begin,
+    PrefixIterator end) {
+  DCHECK(begin != end);
+  size_t count = 0;
+  std::string values;
+  PrefixIterator iter = begin;
+  for (iter = begin;
+       iter != end && count < kMaxInsertRecords;
+       ++count, ++iter) {
+    auto prefix = *iter;
+    DCHECK(prefix.size() >= kHashPrefixSize);
+    std::string hex = base::HexEncode(prefix.data(), kHashPrefixSize);
+    values.append(base::StringPrintf("(x'%s'),", hex.c_str()));
+  }
+  // Remove last comma
+  if (!values.empty()) {
+    values.pop_back();
+  }
+  return {iter, std::move(values), count};
+}
+
+}  // namespace
+
+namespace braveledger_database {
+
+DatabasePublisherPrefixList::DatabasePublisherPrefixList(
+    bat_ledger::LedgerImpl* ledger)
+    : DatabaseTable(ledger) {}
+
+DatabasePublisherPrefixList::~DatabasePublisherPrefixList() = default;
+
+bool DatabasePublisherPrefixList::Migrate(
+    ledger::DBTransaction* transaction,
+    const int target) {
+  DCHECK(transaction);
+  switch (target) {
+    case 28:
+      return MigrateToV28(transaction);
+    default:
+      return true;
+  }
+}
+
+bool DatabasePublisherPrefixList::MigrateToV28(
+    ledger::DBTransaction* transaction) {
+  DCHECK(transaction);
+
+  if (!braveledger_database::DropTable(transaction, kTableName)) {
+    return false;
+  }
+
+  if (!CreateTableV28(transaction)) {
+    return false;
+  }
+
+  ledger_->ClearState(ledger::kStateServerPublisherListStamp);
+  return true;
+}
+
+bool DatabasePublisherPrefixList::CreateTableV28(
+    ledger::DBTransaction* transaction) {
+  DCHECK(transaction);
+  auto command = ledger::DBCommand::New();
+  command->type = ledger::DBCommand::Type::EXECUTE;
+  command->command = base::StringPrintf(
+      "CREATE TABLE %s "
+      "(hash_prefix BLOB PRIMARY KEY NOT NULL)",
+      kTableName);
+
+  transaction->commands.push_back(std::move(command));
+  return true;
+}
+
+void DatabasePublisherPrefixList::Search(
+    const std::string& publisher_key,
+    ledger::SearchPublisherPrefixListCallback callback) {
+  std::string hex = braveledger_publisher::GetHashPrefixInHex(
+      publisher_key,
+      kHashPrefixSize);
+
+  auto command = ledger::DBCommand::New();
+  command->type = ledger::DBCommand::Type::READ;
+  command->command = base::StringPrintf(
+      "SELECT EXISTS(SELECT hash_prefix FROM %s WHERE hash_prefix = x'%s')",
+      kTableName,
+      hex.c_str());
+
+  command->record_bindings = {
+    ledger::DBCommand::RecordBindingType::BOOL_TYPE
+  };
+
+  auto transaction = ledger::DBTransaction::New();
+  transaction->commands.push_back(std::move(command));
+
+  ledger_->RunDBTransaction(std::move(transaction), [callback](
+      ledger::DBCommandResponsePtr response) {
+    if (response && response->result) {
```

Comment:
we should also check for `response->status != ledger::DBCommandResponse::Status::RESPONSE_OK`

Reply:
done.